### PR TITLE
Filling ghost cell in demarcateBrinkmanZone function

### PR DESCRIPTION
**Pull request by Raamkrishnan**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [ ] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

@drwells @amneetb I am filling ghost cells for level set in demarcateBrinkmanZone function in ```BrinkmanPenalizationRigidBodyDynamics``` class. I modified the tests/wave_tank/nwt_cylinder.mpirun=8.output also. Please take a look.

### CODE REVIEW

**Review by amneetb — APPROVED** [review]
@drwells : This was my fault, wherein I was doing ghost cell filling optimization in two routines of BPRigidBodyDyanmics, which ended up being an order-dependent thing.

File: src/IB/BrinkmanPenalizationRigidBodyDynamics.cpp [hunk 1]
```diff
@@ -367,14 +367,29 @@ BrinkmanPenalizationRigidBodyDynamics::demarcateBrinkmanZone(int u_idx, double t
...
+    const int ls_solid_idx = var_db->mapVariableAndContextToIndex(d_ls_solid_var, d_adv_diff_solver->getNewContext());
     const int ls_scratch_idx =
         var_db->mapVariableAndContextToIndex(d_ls_solid_var, d_adv_diff_solver->getScratchContext());
     Pointer<PatchHierarchy<NDIM> > patch_hierarchy = d_adv_diff_solver->getPatchHierarchy();
     int finest_ln = patch_hierarchy->getFinestLevelNumber();
     Pointer<PatchLevel<NDIM> > finest_level = patch_hierarchy->getPatchLevel(finest_ln);
+
+    // Ghost fill the level set values.
+    typedef HierarchyGhostCellInterpolation::InterpolationTransactionComponent InterpolationTransactionComponent;
+    std::vector<InterpolationTransactionComponent> phi_transaction_comps(1);
```
Comment (boyceg) [inline src/IB/BrinkmanPenalizationRigidBodyDynamics.cpp:380]: No need to change, but for this kind of code, initialization lists can make things slightly more compact:
```
    std::vector<InterpolationTransactionComponent> phi_transaction_comps = { ... }
```

**Comment by boyceg** [comment]
If there isn't one already, would it make sense to add a test case for a single processor run as well?

@Raamkrishnan, please also finish up the other items in the checklist before this is merged --- if nothing else, I think that there probably should be a changelog entry for this since it is a bug fix.

**Comment by drwells** [comment]
Thanks for the fix. I'm surprised that this doesn't seem to make a difference for the other tests that use Brinkman penalization.

**Comment by boyceg** [comment]
> Thanks for the fix. I'm surprised that this doesn't seem to make a difference for the other tests that use Brinkman penalization.

This might be the only test that exercises that part of the code. I think that the other Brinkman-related tests are for concentration fields.

**Comment by amneetb** [comment]
> > Thanks for the fix. I'm surprised that this doesn't seem to make a difference for the other tests that use Brinkman penalization.
> 
> This might be the only test that exercises that part of the code. I think that the other Brinkman-related tests are for concentration fields.



> > Thanks for the fix. I'm surprised that this doesn't seem to make a difference for the other tests that use Brinkman penalization.
> 
> This might be the only test that exercises that part of the code. I think that the other Brinkman-related tests are for concentration fields.

Yes.

(We verified with our WEC simulations that the new and old results are pretty much the same)

**Comment by Raamkrishnan** [comment]
I have added a new test for a single processor. The output for both 8 processors and a single processor is the same. I have also added a changelog. Please take a look.

**Comment by amneetb** [comment]
Thanks! Looks good.

**Comment by boyceg** [comment]
Does this fix #1203?

**Comment by Raamkrishnan** [comment]
No. Issue #1203 is related Brinkman advection-diffusion test case which is not related to this bug. BrinkmanPenalizationRigidBodyDynamics class is used to impose solid velocity for the momentum equation.

**Comment by drwells** [comment]
Thanks for adding the serial test - its good to see that we get exactly the same output with 1 or 8 processors.

**Review by drwells — APPROVED** [review]

**Pull request merged**